src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -166,20 +167,30 @@ mavlink_hil_actuator_controls_t Simulator::actuator_controls_from_outputs(const
 	msg.mode |= (armed) ? mode_flag_armed : 0;
 	msg.flags = 0;
 
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+	msg.flags |= mode_flag_lockstep;
+#endif
+
 	return msg;
 }
 
 void Simulator::send_controls()
 {
 	// copy new actuator data if available
 	bool updated = false;
+	bool lockstep = false;
 	orb_check(_actuator_outputs_sub, &updated);
 
-	if (updated) {
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+	// when lockstep is enabled we must send always.
+	lockstep = true;
+#endif
+
+	if (updated || lockstep) {
 		actuator_outputs_s actuators{};
 		orb_copy(ORB_ID(actuator_outputs), _actuator_outputs_sub, &actuators);
 
-		if (actuators.timestamp > 0) {
+		if (actuators.timestamp > 0 || lockstep) {
```

Comment:
I think this was put in order to ignore initial actuator publications without timestamp and no real data. Are you sure that's needed?

Reply:
Yeah, this was me grasping at straws trying to figure out why PX4 stops sending any hil actuator controls.  I'll remove this for now since it is clearly not here that the bug happens, since this change didn't fix it.

Reply:
removed.